lib/api/readable.js

```diff
@@ -267,14 +265,35 @@ function consumeStart (consume) {
   }
 }
 
+/**
+ * @param {Buffer[]} chunks
+ * @param {number} length
+ */
+function chunksDecode (chunks, length) {
+  if (chunks.length === 0 || length === 0) {
+    return ''
+  }
+  const buffer = chunks.length === 1 ? chunks[0] : Buffer.concat(chunks, length)
+
+  const start =
+   buffer.length >= 3 &&
+   // Skip BOM.
+      buffer[0] === 0xef &&
+      buffer[1] === 0xbb &&
+      buffer[2] === 0xbf
+     ? 3
+     : 0
+  return buffer.utf8Slice(start, buffer.length - start)
+}
+
 function consumeEnd (consume) {
   const { type, body, resolve, stream, length } = consume
 
   try {
     if (type === 'text') {
-      resolve(toUSVString(Buffer.concat(body)))
+      resolve(chunksDecode(body, length))
     } else if (type === 'json') {
-      resolve(JSON.parse(Buffer.concat(body)))
+      resolve(chunksDecode(body, length))
```

Comment:
So here the `JSON.parse` was irrelevant? Wow.

Reply:
Are we missing a test?

Reply:
Sorry, it was accidentally deleted.